halley-core/src/main/kotlin/com/infinum/halley/core/Halley.kt

```diff
@@ -56,7 +58,7 @@ public class Halley(
     }
 
     init {
-        HttpClientCache.save(httpClient)
+        CallFactoryCache.save(callFactory)
```

Comment:
Meaning, this might be a good place to wrap the client so everywhere we have the wrapped CF.
```suggestion
        CallFactoryCache.save(callFactory.asyncCallFactory())
```